runtime/common/src/migrations/transfer_allowlist_currency.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::{AccountId, BlockNumber};
+use cfg_types::{
+	locations::Location,
+	tokens::{CurrencyId, FilterCurrency},
+};
+use frame_support::{
+	pallet_prelude::{NMapKey, OptionQuery},
+	storage::types::{StorageDoubleMap, StorageNMap},
+	traits::{Get, OnRuntimeUpgrade, StorageInstance},
+	weights::Weight,
+	Blake2_128Concat, Twox64Concat,
+};
+use pallet_transfer_allowlist::{AllowanceDetails, AllowanceMetadata};
+#[cfg(feature = "try-runtime")]
+use parity_scale_codec::{Decode, Encode};
+#[cfg(feature = "try-runtime")]
+use sp_runtime::DispatchError;
+
+const LOG_PREFIX: &str = "TransferAllowlist-CurrencyMigration: ";
+
+struct DelayPrefix;
+impl StorageInstance for DelayPrefix {
+	const STORAGE_PREFIX: &'static str = "AccountCurrencyTransferCountDelay";
+
+	fn pallet_prefix() -> &'static str {
+		"TransferAllowList"
+	}
+}
+type OldAccountCurrencyTransferCountDelay = StorageDoubleMap<
+	DelayPrefix,
+	Twox64Concat,
+	AccountId,
+	Twox64Concat,
+	CurrencyId,
+	AllowanceMetadata<BlockNumber>,
+	OptionQuery,
+>;
+
+struct AllowancePrefix;
+impl StorageInstance for AllowancePrefix {
+	const STORAGE_PREFIX: &'static str = "AccountCurrencyTransferAllowance";
+
+	fn pallet_prefix() -> &'static str {
+		"TransferAllowList"
+	}
+}
+type OldAccountCurrencyTransferAllowance = StorageNMap<
+	AllowancePrefix,
+	(
+		NMapKey<Twox64Concat, AccountId>,
+		NMapKey<Twox64Concat, CurrencyId>,
+		NMapKey<Blake2_128Concat, Location>,
+	),
+	AllowanceDetails<BlockNumber>,
+	OptionQuery,
+>;
+
+pub struct Migration<T>(sp_std::marker::PhantomData<T>);
+impl<
+		T: pallet_transfer_allowlist::Config<
+			AccountId = AccountId,
+			Location = Location,
+			CurrencyId = FilterCurrency,
+			BlockNumber = BlockNumber,
+		>,
+	> OnRuntimeUpgrade for Migration<T>
+{
+	fn on_runtime_upgrade() -> Weight {
+		log::info!("{LOG_PREFIX} Migrating currency used started...");
```

Comment:
NIT: Since this is a breaking storage change, we should bump the `StorageVersion` of `pallet-transfer-allowlist` and check versions. Normally, I would have flagged this as `CR` but no reason to be that pedantic here for the sake of the 1025 deadline.

Reply:
Will still do.

Reply:
Changed